src/Sets/SparsePolynomialZonotope/linear_map.jl

```diff
@@ -0,0 +1,82 @@
+"""
+	linear_map(MZ::MatrixZonotope, P::SparsePolynomialZonotope)
+
+Compute the linear map of a sparse polynomial zonotope through a matrix zonotope,
+following Proposition 1 of [HuangLBS2025](@citet).
+
+### Input
+
+- `lm` -- a linear map of a sparse polynomial zonotope through a matrix zonotope
+
+### Output
+
+A sparse polynomial zonotope representing the linear map ``MZ ⋅ P```.
+
+"""
+function linear_map(MZ::MatrixZonotope, P::SparsePolynomialZonotope)
+    if ngens_indep(P) > 0
+        error("an exact expression for the linear map is only available for " *
+              "`SparsePolynomialZonotope`s with no independent generators. " *
+              "Try using `overapproximate` instead")
+    end
+
+    @assert size(MZ, 2) == dim(P) "a linear map of size $(size(M)) cannot " *
+                                  "be applied to a set of dimension $(dim(X))"
```

Comment:
There is now a new convenient way to validate the input via `@validate`.

```suggestion
@validate function linear_map(MZ::MatrixZonotope, P::SparsePolynomialZonotope)
    if ngens_indep(P) > 0
        error("an exact expression for the linear map is only available for " *
              "`SparsePolynomialZonotope`s with no independent generators. " *
              "Try using `overapproximate` instead")
    end
```

Reply:
Sorry, there is a loading dependency. Not so easy to resolve. Maybe undo the change and remove `@validate` again (you may be able to call `validate_linear_map`, or just bring back your old assertion).

Reply:
I brought back the assertion for now